runelite-client/src/main/java/net/runelite/client/plugins/microbot/util/walker/Rs2Walker.java

```diff
@@ -296,11 +296,19 @@ private static boolean handleDoors(WorldPoint currentWorldPoint, WorldPoint next
         if (wallObject != null) {
             ObjectComposition objectComposition = Rs2GameObject.getObjectComposition(wallObject.getId());
 
-            for (var action : objectComposition.getActions()) {
-                if (action != null && action.contains("Pay-toll")) {
-                    Rs2GameObject.interact(wallObject, action);
-                    Rs2Player.waitForWalking();
-                    return true;
+            for (var action : objectComposition.getActions()){
+                if(action != null){
```

Comment:
if (action != null && (action.contains("Pay-toll") || action.contains("Pick-lock"))

Seems cleaner and you will avoid duplicate code

Reply:
I have also simplified the logic here on commit d9572de